plugins/trace_api_plugin/store_provider.cpp

```diff
@@ -98,6 +98,43 @@ namespace eosio::trace_api {
       return std::make_tuple( entry.value(), irreversible );
    }
 
+   // Returns true if `block` includes the transaction identified by `trx_id`
+   static bool block_has_trx(const get_block_t& block, const chain::transaction_id_type& trx_id) {
+      bool           has_trx     = false;
+      data_log_entry block_trace = std::get<0>(*block);
+
+      if (std::holds_alternative<block_trace_v0> (block_trace)) {
+         auto& trace_v0 = std::get<block_trace_v0>(block_trace);
+         auto  itr      = std::find_if(trace_v0.transactions.begin(), trace_v0.transactions.end(),
+                                       [&](const auto& trx) { return trx.id == trx_id; });
+         has_trx = (itr != trace_v0.transactions.end());
+      } else if (std::holds_alternative<block_trace_v1>(block_trace)) {
+         auto& trace_v1 = std::get<block_trace_v1>(block_trace);
+         auto  itr      = std::find_if(trace_v1.transactions_v1.begin(), trace_v1.transactions_v1.end(),
+                                       [&](const auto& trx) { return trx.id == trx_id; });
+         has_trx        = (itr != trace_v1.transactions_v1.end());
+      } else if (std::holds_alternative<block_trace_v2>(block_trace)) {
+         auto& trace_v2 = std::get<block_trace_v2>(block_trace);
+         if (std::holds_alternative<std::vector<transaction_trace_v2>>(trace_v2.transactions)) {
+            auto& trxs = std::get<std::vector<transaction_trace_v2>>(trace_v2.transactions);
+            auto  itr  = std::find_if(trxs.begin(), trxs.end(),
+                                      [&](const auto& trx) { return trx.id == trx_id; });
+            has_trx    = (itr != trxs.end());
+         } else if (std::holds_alternative<std::vector<transaction_trace_v3>>(trace_v2.transactions)) {
+            auto& trxs = std::get<std::vector<transaction_trace_v3>>(trace_v2.transactions);
+            auto  itr  = std::find_if(trxs.begin(), trxs.end(),
+                                      [&](const auto& trx) { return trx.id == trx_id; });
+            has_trx    = (itr != trxs.end());
+         } else {
+            FC_ASSERT( false, "block_trace_v2 should contain transaction_trace_v2 or transaction_trace_v3 transactions" );;
+         }
+      } else {
+         FC_ASSERT( false, "block_trace should be a block_trace_v0, block_trace_v1, or a block_trace_v3" );
+      }
+
+      return has_trx;
+   }
```

Comment:
I can't really tell for sure, but would a `visit()` + overloaded lambda be more concise or cleaner?

Reply:
The function is no longer needed.